### PR TITLE
[Feature] Allow customizing title and subtitle

### PR DESCRIPTION
**Pull request by jpeng-ms**
## Purpose
This PR adds a new ability for Contoso developer to customize the title and subtitle on set up view. 

## Does this introduce a breaking change?

```
[] Yes
[X] No
```

new options:
```
launch() -> 
    - remote options
    - local options
        - participant view data
        - navigation bar view data      <- NEW
            - title                     <- NEW
            - subtitle                  <- NEW
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
1. download the code, run it in simulator
2. tap on settings, enter title and subtitle
3. close settings, tap on start experience 

## What to Check
1. make sure title and subtitle are reflecting what you have entered in settings
2. if title is empty, check if default title "setup" was used and subtitle is hidden
3. if subtitle is empty, check if subtitle is hidden
4. title's colour is "text dominant" for both light/dark modes, subtitle's colour is "text secondary" for light mode and "text dominant" for dark mode.

## Other Information
1. checked with UI Design, contoso developer is able to enter unlimited characters, but we would truncate the title/subtitle when they become too long
2. Unit tests is coming up as a second PR for this ticket.3. 
3. Android PR: https://github.com/Azure/communication-ui-library-android/pull/383
### happy path
image
image

### edge cases
image
image

image

Simulator Screen Shot - iPad Pro (11-inch) (3rd generation) - 2022-08-15 at 14 53 07

### CODE REVIEW

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift [hunk 1]
```diff
@@ -9,12 +9,17 @@ import UIKit
...
+    let participantViewData: ParticipantViewData?
+    /// The NavigationBarViewData would be used to populate title and subtitle on setup view
+    let navigationBarViewData: NavigationBarViewData?
     /// Create an instance of LocalOptions. All information in this object is only stored locally in the composite.
     /// - Parameters:
     ///    - participantViewData: The ParticipantViewData to be displayed for local participants avatar
-    public init(participantViewData: ParticipantViewData) {
+    ///    - navigationBarViewData: The NavigationBarViewData to be shown on navigation bar of set up view
+    public init(participantViewData: ParticipantViewData? = nil,
+                navigationBarViewData: NavigationBarViewData? = nil) {
```
Comment (Leah-Xia-Microsoft) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift:20]: Do we need to set the default value to nil for an optional parameter?

**Review by Leah-Xia-Microsoft — DISMISSED** [review]
LGTM. Tested on iPhone 13

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift (hunk 1 above)
Comment (jpeng-ms) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift:20]: yes! otherwise it would actually create a compile error on Contoso side and upgrading this ticket to "breaking changes". For participant view data, it should also have a default nil so that both of them are completely optional and contoso can decide what they wanted to do without having a compile error.

**Review by JoshuaLai — DISMISSED** [review]

**Comment by jpeng-ms** [comment]
resolved a merge conflict in CHANGELOG, please review again

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift [hunk 2]
```diff
@@ -9,12 +9,17 @@ import UIKit
...
-    let participantViewData: ParticipantViewData
+    let participantViewData: ParticipantViewData?
+    /// The NavigationBarViewData would be used to populate title and subtitle on setup view
+    let navigationBarViewData: NavigationBarViewData?
     /// Create an instance of LocalOptions. All information in this object is only stored locally in the composite.
     /// - Parameters:
     ///    - participantViewData: The ParticipantViewData to be displayed for local participants avatar
-    public init(participantViewData: ParticipantViewData) {
+    ///    - navigationBarViewData: The NavigationBarViewData to be shown on navigation bar of set up view
+    public init(participantViewData: ParticipantViewData? = nil,
```
Comment (emlynmac) [inline AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift:19]: Are we sure we want to create this breaking change?

**Review by jimchou-dev — APPROVED** [review]
Tested, LGTM

**Review by Leah-Xia-Microsoft — APPROVED** [review]

File: AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift (hunk 2 above)
Comment (jpeng-ms) [reply AzureCommunicationUI/sdk/AzureCommunicationUICalling/Sources/CallCompositeOptions/LocalOptions.swift:19]: Hi @emlynmac, this won't be a breaking change right? we are downgrading a required parameter to optional. If Contoso developer had parameter defined in their app right now, the local option would continue to work as normal and no compile errors.
